swift/ios-shoppe-demo/Views/CheckoutTextViewCell.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  CheckoutTextViewCell.swift
+//  ios-shoppe-demo
+//
+//  Created on 4/10/20.
+//  Copyright © 2020 FullStory All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class CheckoutTableViewCell: UITableViewCell, UITextFieldDelegate {
+    @IBOutlet weak var addressDetailTextField: UITextField!
+
+    var tableViewController: CheckoutTableViewController?
+    var addressDetail: AddressDetail? {
+        didSet {
+            setup()
```

Comment:
A `setup` feels like a constructor (one time) operation.  This code executes `setup` on every assignment.  `setup` is used to show a tooltip if no content is entered, correct?  If so, we might change `setup` to be `showTooltip` or something more descriptive.  If the Detail textfields get typed in and then have an empty string, does the tooltip get shown?